### DISCUSSION TITLE
How to embed AvaloniaResource?

### DISCUSSION THREAD

**Discussion by NotroDev** [AvaloniaUI/Avalonia#13345] [Q&A] [answered]
Hello,

I have a folder called "Languages" in my project. Each `.axaml` file in this folder contains translations.
However, I have a problem. To use these files, they need to be defined as `AvaloniaResource`s and loaded like this:

```cs
var langResource = new ResourceInclude(new Uri($"avares://MyProject/Languages/{language}.axaml"))
{
	Source = new Uri($"avares://MyProject/Languages/{language}.axaml")
};
Application.Current.Resources.MergedDictionaries.Add(langResource);
```

I would like the "Languages" folder and its files to be included in the compiled program directory so that users can edit these files, and any changes they make will take effect. In WPF, the `EmbeddedResource` approach worked, but I don't think I can use a file with a type other than `AvaloniaResource` in `ResourceInclude` in Avalonia.

**Comment by timunie** [comment]
https://github.com/timunie/Tims.Avalonia.Samples

Check Localization sample

**Reply by NotroDev** [reply]
~~Oh. A moment before posting this I checked your sample, but I thought it doesn't have it.
Thanks.~~

Correct response below

**Reply by NotroDev** [reply]
Oops, sorry, I'm not thinking after the whole day.
I don't think in your sample is how I want it to be.
I can edit files in the `Assets/Lang` folder, but after compiling program, there is no `Assets` folder.

**Reply by timunie** [reply]
Needs to be avaloniaResource

**Reply by timunie** [reply]
You can cooy the files to the LangFolder, but no need to do iirc

**Reply by NotroDev** [reply]
Hm? It is AvaloniaResource.
Only EmbeddedResource uses files from the compiled program's folder, AvaloniaResource uses files from `.dll` or whatever.

**Reply by jp2masa** [reply]
I think both `AvaloniaResource` and `EmbeddedResource` embed the files in the `.dll`.

**Reply by workgroupengineering** [reply]
`AvaloniaResource` is parsed from `XamlC` and than embedded

`EmbeddedResource` is embedded as is

**Reply by NotroDev** [reply]
Okay, but does someone know how to do it? 😅

**Reply by workgroupengineering** [reply]
## AvaloniaResource
- Generally all [a]xaml files (perf reason)
- when you need to invoke the resource via the  schema `avares://` (eg: use of fonts or images in the xaml)

## EmbeddedResource 
- all other cases

**Reply by NotroDev** [reply]
Thanks, but I need these AvaloniaResource files to be in **compiled program's folder**, and if user edits these files, program uses files with those changes, and doesn't reset the files.

Normally, these files can only be edited before compilation.

**Reply by workgroupengineering** [reply, marked as answer]
in this case neither of them is suitable. You need to do parsing at runtime. You should use `Content` as `BuildAction` and always enable copy.

To load the xaml you should use `AvaloniaRuntimeXamlLoader.Load`. Personally I would recommend you use `json` instead of xaml.

**Reply by NotroDev** [reply]
Okay, thanks! I'll try it.

**Reply by NotroDev** [reply]
Yes, it works! Thank you so much, @workgroupengineering!
